### PR TITLE
SEP-38: integrate with the anchor synchronous callback API for `GET /rate`

### LINKED ISSUE #76: SEP-38: mock calls to the Anchor `GET /rate?type=indicative` and `GET /rate?type=firm` and save the returned firm quote in the Platform database.

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Integrate with the anchor synchronous callback API for `GET /rate`.

### Why

Closes #76

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/integration/rate/GetRateRequest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package org.stellar.anchor.integration.rate;
+
+import com.google.gson.annotations.SerializedName;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class GetRateRequest {
```
Comment (JakeUrban) [inline core/src/main/java/org/stellar/anchor/integration/rate/GetRateRequest.java:9]: 🤔 I think @lijamie98 wanted to merge the request/response classed defined in `anchor/integration` with `anchor/dto`, but I'm not sure when we wanted to implement that. Just a note.

File: platform/src/main/java/org/stellar/anchor/platform/IntegrationConfig.java [hunk 2]
```diff
@@ -15,4 +17,10 @@ CustomerIntegration customerIntegration(Sep12Config sep12Config) {
     return new PlatformCustomerIntegration(
         sep12Config.getCustomerIntegrationEndPoint(), httpClient);
   }
+
+  @Bean
+  RateIntegration rateIntegration(Sep38Config sep38Config) {
+    OkHttpClient httpClient = OkHttpUtil.buildClient();
```
Comment (JakeUrban) [inline platform/src/main/java/org/stellar/anchor/platform/IntegrationConfig.java:23]: 💡 When I was implementing SEP-12's integration I was thinking that we should at some point update all our clients to use sensible timeout / retry / connection pool policies.

Not blocking for now though.

File: platform/src/main/java/org/stellar/anchor/platform/PlatformRateIntegration.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  public GetRateResponse getRate(GetRateRequest request) throws AnchorException {
+    Builder urlBuilder = get(anchorEndpoint).newBuilder().addPathSegment("rate");
+    Type type = new TypeToken<Map<String, ?>>() {}.getType();
+    Map<String, String> paramsMap = gson.fromJson(gson.toJson(request), type);
+    paramsMap.forEach(
+        (key, value) -> {
+          if (value != null) {
+            urlBuilder.addQueryParameter(key, value);
+          }
+        });
```
Comment (JakeUrban) [inline platform/src/main/java/org/stellar/anchor/platform/PlatformRateIntegration.java:46]: 👏 This is nice, I didn't know of a way to take an object and flexibly turn it into a map.

**Review by JakeUrban — APPROVED** [review]
Awesome. I like the patterns that are developing between core<-->platform for each SEP. Glad to see you were able to move some of the SEP-12 code to common logic too.

File: platform/src/main/java/org/stellar/anchor/platform/IntegrationConfig.java (hunk 2 above)
Comment (marcelosalloum) [reply platform/src/main/java/org/stellar/anchor/platform/IntegrationConfig.java]: That makes sense to me too. I created a bean for the HttpClient in https://github.com/stellar/java-stellar-anchor-sdk/pull/108/commits/e0741bd56fd7c894b13feb0231c2d498485e7935.

**Pull request merged**